frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -295,6 +320,46 @@ pub mod pallet {
             Ok(().into())
         }
 
+        /// add contract address to the preaproved list.
+        /// contract_id should be ink! or evm contract.
+        ///
+        /// Sudo call is required
+        #[pallet::weight(1_000_000)]
+        pub fn contract_pre_approval(
+            origin: OriginFor<T>,
+            contract_id: SmartContract<T::AccountId>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            ensure!(
+                !RegisteredDapps::<T>::contains_key(&contract_id),
+                Error::<T>::AlreadyRegisteredContract
+            );
+
+            let mut pre_approved_contracts = Self::pre_approved_contracts();
+            ensure!(
+                !pre_approved_contracts.contains(&contract_id),
+                Error::<T>::AlreadyPreApprovedContract
+            );
+            pre_approved_contracts.push(contract_id);
+            PreApprovedContracts::<T>::put(pre_approved_contracts);
+
+            Ok(().into())
+        }
+
+        /// Enable or disable adding new contracts to the pre-approved list
+        ///
+        /// Sudo call is required
+        #[pallet::weight(1_000_000)]
+        pub fn preaproval_for_new_contact_enabled(
```

Comment:
spelling issue: `preaproval` -> `preapproval`

Most functions and variables use `pre-approval` while this function is `preapproval` as a single word. Let's stick to one.

Also, I think the function name should be something easier like `enable_contract_preapproval` (or `pre_approval`).

Reply:
done